3/feat/iconcomponent

dominionworkz opened the pull request:
Closes #3 
PR was created to verify if the codebase is good to go or if there's a need for revisions:

Create an Icon component that:

1. Accepts a name prop that is required.
2. Accepts a size prop that is optional. The default value should be 24.
3. Returns an <svg> with the given name and size. Returns null if no name is given.

dominionworkz committed:
feat: Create Icon component (#3)

geraldiner commented on client/src/components/Icon.jsx:
```diff
@@ -0,0 +1,21 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+function Icon({ id, size = 24 }) {
```
Since we're using `prop-types` now, I think we don't have to set the default up here. Try removing the `= 24` part and seeing if it still works since `Icon.defaultProps` for `size` is listed below.

geraldiner reviewed (commented):
Nice first "real" PR 🥳 

One small nit, and I didn't see an implementation of this part from the third bullet:

> Returns null if no name is given.

dominionworkz commented:
This is what you were waiting for me to comment on correct? specifically about the removal of =24 ?

geraldiner commented:
> One small nit, and I didn't see an implementation of this part from the third bullet:
> 
> > Returns null if no name is given.

Also this

dominionworkz commented:
The null should show where the icons would show correct?

On Tue, Jan 23, 2024, 5:54 AM Geraldine ***@***.***> wrote:

> One small nit, and I didn't see an implementation of this part from the
> third bullet:
>
> Returns null if no name is given.
>
> Also this
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/geraldiner/advent-of-js-and-css-2023/pull/48#issuecomment-1905789532>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJBBYXMFYYIBACUR5STPZ5DYP6JIHAVCNFSM6AAAAABB5TV3RKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBVG44DSNJTGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

geraldiner commented:
Not exactly. Look up what `null` means

On Tue, Jan 23, 2024, 8:45 AM Quincy Bulla ***@***.***> wrote:

> The null should show where the icons would show correct?
>
> On Tue, Jan 23, 2024, 5:54 AM Geraldine ***@***.***> wrote:
>
> > One small nit, and I didn't see an implementation of this part from the
> > third bullet:
> >
> > Returns null if no name is given.
> >
> > Also this
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/geraldiner/advent-of-js-and-css-2023/pull/48#issuecomment-1905789532>,
>
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AJBBYXMFYYIBACUR5STPZ5DYP6JIHAVCNFSM6AAAAABB5TV3RKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBVG44DSNJTGI>
>
> > .
> > You are receiving this because you authored the thread.Message ID:
> > ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/geraldiner/advent-of-js-and-css-2023/pull/48#issuecomment-1906086576>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB3BAKQLCUPQCQRFJQB5CNLYP65GVAVCNFSM6AAAAABB5TV3RKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBWGA4DMNJXGY>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

dominionworkz commented:
I hope this is correct now, lol

geraldiner commented on client/src/components/Icon.jsx:
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+function Icon({ id, size }) {
+  return (
```
```suggestion
  if (!id) {
    return null;
  }
  return (
```

geraldiner replied:
This is all I meant by returning null if there's no `id` given.

geraldiner commented:
> I hope this is correct now, lol

Not quite. Also, if you want a re-review, you can click the 🔄 icon next to my name under "Reviewers" on the side

geraldiner commented:
@dominionworkz Did you actually make any changes since the last review or were you just trying out the "Request re-review" button?

dominionworkz commented:
> @dominionworkz Did you actually make any changes since the last review or were you just trying out the "Request re-review" button?

I'm did try to make change & the same with the button.  I did not see if anything occurred

geraldiner commented on client/src/components/Icon.jsx:
```diff
@@ -4,17 +4,18 @@ import React from 'react';
 
 Icon.propTypes = {
-  id: PropTypes.string.isRequired,
+  id: PropTypes.string,
```
The `id` prop is supposed to be required. I think you should put this back

geraldiner commented on client/src/components/Icon.jsx:
```diff
@@ -4,17 +4,18 @@ import React from 'react';
 };
 
 Icon.defaultProps = {
+  id: null,
```
I think this is unnecessary

geraldiner commented on client/src/components/Icon.jsx:
```diff
@@ -4,17 +4,18 @@ import React from 'react';
   return (
     <svg viewBox="0 0 24 24" width={size} height={size}>
-      <use href={`/images/icons/sprite.svg#${id}`} />
+      <use href={`/images/icons/sprite.svg#${id?.toString()}`} />
```
What does this add to the code?
